implementations/rust/ockam/ockam_vault/src/secret_impl.rs

```diff
@@ -41,6 +41,41 @@ impl SoftwareVault {
             SecretType::Buffer | SecretType::Aes | SecretType::P256 => None,
         })
     }
+
+    /// Validate secret key.
+    pub fn check_secret(
+        &mut self,
+        secret: &[u8],
+        attributes: &SecretAttributes,
+    ) -> ockam_core::Result<()> {
+        match attributes.stype() {
+            SecretType::Curve25519 => {
+                if secret.len() != CURVE25519_SECRET_LENGTH {
+                    return Err(VaultError::InvalidCurve25519SecretLength.into());
+                }
+                let clamped = x25519_dalek::StaticSecret::from(
+                    TryInto::<[u8; CURVE25519_SECRET_LENGTH]>::try_into(secret).unwrap(),
```

Comment:
If this conversion succeeded, then secret.len() == CURVE25519_SECRET_LENGTH. So, instead of unwrap we can return InvalidCurve25519SecretLength if conversion failed, and remove the length check above